devtools/enforcer-rules/src/main/java/io/quarkus/enforcer/RequiresMinimalDeploymentDependency.java

```diff
@@ -0,0 +1,184 @@
+        //   or those parents are not relevant
+
+        Map<String, Artifact> nonDeploymentArtifactsByGAV = project.getArtifacts().stream()
+                .filter(artifact -> "jar".equals(artifact.getType()))
+                .filter(artifact -> artifact.getGroupId().startsWith(GROUP_ID_PREFIX))
+                .filter(artifact -> !artifact.getArtifactId().endsWith(DEPLOYMENT_ARTIFACT_ID_SUFFIX))
+                .collect(Collectors.toMap(this::buildGAVKey, a -> a));
+
+        // Skip if artifacts are not resolved.
+        // To avoid this "soft exit", explicit resolving would be necessary but that is pretty elaborate in an enforcer rule.
+        // If the build goal is "late" enough, artifacts for the respective scope *will* be resolved automatically.
+        if (nonDeploymentArtifactsByGAV.values().stream().anyMatch(artifact -> !artifact.isResolved())) {
+            logger.warn("Skipping rule " + RequiresMinimalDeploymentDependency.class.getSimpleName()
+                    + ": Artifacts are not resolved, consider using a later build goal like 'package'.");
+            return;
+        }
+
+        String projArtifactKey = buildGAVKey(project.getArtifact());
+
+        Map<String, Dependency> directDepsByGAV = project.getDependencies().stream()
+                .filter(d -> d.getGroupId().startsWith(GROUP_ID_PREFIX))
+                .collect(Collectors.toMap(d -> d.getGroupId() + ":" + d.getArtifactId() + ":" + d.getVersion(), d -> d,
+                        (a, b) -> a));
+
+        List<String> missingDeploymentDeps = nonDeploymentArtifactsByGAV.entrySet().parallelStream()
+                .filter(entry -> directDepsByGAV.containsKey(entry.getKey())) // only direct deps
+                .map(entry -> DEPLOYMENT_GAV_CACHE.computeIfAbsent(entry.getKey(), k -> parseDeploymentGAV(entry.getValue())))
+                .filter(optDeploymentGAV -> optDeploymentGAV
+                        .map(deploymentGAV -> !isMinDeploymentDepPresent(deploymentGAV, projArtifactKey, directDepsByGAV))
+                        .orElse(false))
+                .map(Optional::get)
+                .sorted()
+                .collect(Collectors.toList());
+
+        if (!missingDeploymentDeps.isEmpty()) {
+            String requiredDeps = missingDeploymentDeps.stream()
+                    .map(gav -> (Object[]) gav.split(":"))
+                    .map(gavArray -> String.format(DEP_TEMPLATE, gavArray))
+                    .collect(Collectors.joining("\n"));
+            throw new EnforcerRuleException(missingDeploymentDeps.size()
+                    + " *-deployment dependencies are missing/configured incorrectly:\n"
+                    + "    " + missingDeploymentDeps.stream().collect(Collectors.joining("\n    "))
+                    + "\n\nTo fix this issue, add the following dependencies to pom.xml:\n\n"
+                    + "        <!-- \"Symbolic\" test dependencies to *-deployment artifacts for consistent build order -->\n"
```

Comment:
I'd just remove the `"Symbolic"` part, they are in no way symbolic, they are perfectly real :-) And their purpose is explained by "for consistent build order". We could add something like `(not needed for any other reason)` to the end, but I think that wouldn't be necessary.

Reply:
Glad you see it this way because I was not entirely happy with "symbolic" vs./+ "minimal" either.

How about replacing "Symbolic" with "Minimal", hinting at the rule class and adding some javadoc to the rule?

Reply:
> but I assume those changes are just copy&paste of the enforcer output, so should be pretty safe.

They definitely are. 😉

Reply:
Minimal makes a lot more sense, as the enforcer rule attempts to find a minimal set of `-deployment` dependencies to add. I personally don't think it's necessary, but I'll leave it to you :-)

Reply:
Done.